### PR TITLE
Introducing Hooks Spanish translation

### PR DESCRIPTION
**Pull request by KevinGomezDev**
Since there's not a current translation for Hooks FAQ, i used [Preguntas frecuentes acerca de Hooks] as a translation for Hooks FAQ

### CODE REVIEW

File: content/docs/hooks-intro.md [hunk 1]
```diff
@@ -1,112 +1,112 @@
...
+También entendemos que la barra para añadir un nuevo primitivo a React es extremadamente alta. Para los lectores curiosos, hemos preparado un,[RFC detallado](https://github.com/reactjs/rfcs/pull/68) que se sumerge en la motivación con más detalles, y proporciona una perspectiva extra sobre las decisiones de diseño específicas y el estado de la técnica relacionado.
 
-**Crucially, Hooks work side-by-side with existing code so you can adopt them gradually.** We are sharing this experimental API to get early feedback from those in the community who are interested in shaping the future of React — and we will iterate on Hooks in the open.
+**Es crucial, que los Hooks trabajen codo a codo con el código existente para que puedas adoptarlos gradualmente.** Estamos compartiendo esta API experimental para obtener retroalimentación temprana de aquellos en la comunidad que están interesados en dar forma al futuro de React - y vamos a iterar en Hooks al aire libre.
```
Comment (icarlossz) [inline content/docs/hooks-intro.md]: I'd put `de forma abierta` instead `al aire libre`.

File: content/docs/hooks-intro.md [hunk 2]
```diff
@@ -1,112 +1,112 @@
...
+## Sin Cambios Disruptivos {#no-breaking-changes}
 
-Before we continue, note that Hooks are:
+Antes de continuar, debes notar que los Hooks son:
 
-* **Completely opt-in.** You can try Hooks in a few components without rewriting any existing code. But you don't have to learn or use Hooks right now if you don't want to.
-* **100% backwards-compatible.** Hooks don't contain any breaking changes.
-* **Available now.** Hooks are now available with the release of v16.8.0.
+* **Completamente opcionales.** Puedes probar Hooks en unos pocos componentes sin re-escribir ningún código existente. Pero no tienes que aprender o usar Hooks ahora mismo si no quieres.
+* **100% compatibles con versiones anteriores.** Hooks no contienen cambios disruptivos a las versiones existentes.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
* **100% compatibles con versiones anteriores.** Los Hooks no tienen cambios con rupturas con respecto a versiones existentes.
```

File: content/docs/hooks-intro.md [hunk 3]
```diff
@@ -1,112 +1,112 @@
...
+* **Disponibles de inmediato.** Hooks ahora están disponibles con el lanzamiento de la versión v16.8.0.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
* **Disponibles de inmediato.** Los Hooks ya están disponibles con el lanzamiento de la versión v16.8.0.
```

File: content/docs/hooks-intro.md [hunk 4]
```diff
@@ -1,112 +1,112 @@
...
 
-**There are no plans to remove classes from React.** You can read more about the gradual adoption strategy for Hooks in the [bottom section](#gradual-adoption-strategy) of this page.
+**No hay planes para remover el módelo de clases de React.** Puedes leer más sobre la estrategia de adopción gradual de Hooks en la [sección inferior](#gradual-adoption-strategy) de esta página.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
**No hay planes para remover el modelo de clases de React.** Puedes leer más sobre la estrategia de adopción gradual de Hooks en la [sección inferior](#gradual-adoption-strategy) de esta página.
```

File: content/docs/hooks-intro.md [hunk 5]
```diff
@@ -1,112 +1,112 @@
...
 
-**Hooks don't replace your knowledge of React concepts.** Instead, Hooks provide a more direct API to the React concepts you already know: props, state, context, refs, and lifecycle. As we will show later, Hooks also offer a new powerful way to combine them.
+**Hooks no reemplazan tu conocimiento de los conceptos de React.** Por el contrario, los Hooks proporcionan una API más directa a los conceptos que ya conoces de React: _props_, estado, context, refs, y ciclo de vida. Como veremos más adelante, los hooks también ofrecen una nueva y poderosa forma de combinarlos.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
**Los Hooks no reemplazan tu conocimiento de los conceptos de React.** Por el contrario, los Hooks proporcionan una API más directa a los conceptos que ya conoces de React: _props_, estado, contexto, referencias, y ciclo de vida. Como veremos más adelante, los hooks también ofrecen una nueva y poderosa forma de combinarlos.
```

File: content/docs/hooks-intro.md [hunk 6]
```diff
@@ -1,112 +1,112 @@
...
 
-**If you just want to start learning Hooks, feel free to [jump directly to the next page!](/docs/hooks-overview.html)** You can also keep reading this page to learn more about why we're adding Hooks, and how we're going to start using them without rewriting our applications.
+**Si sólo quieres empezar a aprender a usar hooks, ¡no dudes en [saltar directamente a la página siguiente!](/docs/hooks-overview.html)** También puedes seguir leyendo esta página para saber más acerca de por qué estamos añadiendo Hooks y cómo vamos a empezar a usarlos sin tener que reescribir nuestras aplicaciones.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
**Si sólo quieres empezar a aprender a usar hooks, ¡no dudes en [saltar directamente a la siguiente página!](/docs/hooks-overview.html)** También puedes seguir leyendo esta página para saber más acerca de por qué estamos añadiendo Hooks y cómo vamos a empezar a usarlos sin tener que reescribir nuestras aplicaciones.
```

File: content/docs/hooks-intro.md [hunk 7]
```diff
@@ -1,112 +1,112 @@
...
 
-## Motivation {#motivation}
+## Motivación {#motivation}
 
-Hooks solve a wide variety of seemingly unconnected problems in React that we've encountered over five years of writing and maintaining tens of thousands of components. Whether you're learning React, use it daily, or even prefer a different library with a similar component model, you might recognize some of these problems.
+Los Hooks resuelven una amplia variedad de problemas aparentemente desconectados en React que hemos encontrado durante más de cinco años de escribir y mantener decenas de miles de componentes. Ya sea que estés aprendiendo React, usandolo diariamente o incluso prefieras una librería diferente con un modelo de componentes similar, es posible que reconozcas algunos de estos problemas.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
Los Hooks resuelven una amplia variedad de problemas aparentemente desconectados en React que hemos encontrado durante más de cinco años de escribir y mantener decenas de miles de componentes. Ya sea que estés aprendiendo React, usándolo diariamente o incluso prefieras una librería diferente con un modelo de componentes similar, es posible que reconozcas algunos de estos problemas.
```

File: content/docs/hooks-intro.md [hunk 8]
```diff
@@ -1,112 +1,112 @@
...
+RReact no ofrece una forma de "acoplar" comportamientos reutilizables a un componente (Por ejemplo, al conectarse a un _store_). Si llevas un tiempo trabajando con React, puedes estar familiarizado con patrones como [render props](/docs/render-props.html) y [componentes de alto orden](/docs/higher-order-components.html) que tratan resolver esto. Pero estos patrones requieren que reestructures tus componentes al usarlos, lo cual puede ser complicado y hacen que tu código sea más difícil de seguir. Si observas una aplicación típica de React usando _React DevTools_, Lo más probable es que encuentres un "wrapper hell" de componentes envueltos en capas de _providers_, _consumers_, _componentes de alto orden_, _render props_, y otras abstracciones. Aunque podemos [filtrarlos usando las DevTools](https://github.com/facebook/react-devtools/pull/503), esto apunta a un problema aún más profundo: React necesita un mejor primitivo para compartir lógica de estado.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
React no ofrece una forma de "acoplar" comportamientos re-utilizables a un componente (por ejemplo, al conectarse a un _store_). Si llevas un tiempo trabajando con React, puedes estar familiarizado con patrones como [render props](/docs/render-props.html) y [componentes de orden superior](/docs/higher-order-components.html) que tratan de resolver esto. Pero estos patrones requieren que reestructures tus componentes al usarlos, lo cual puede ser complicado y hacen que tu código sea más difícil de seguir. Si observas una aplicación típica de React usando _React DevTools_, Lo más probable es que encuentres un "wrapper hell" de componentes envueltos en capas de _providers_, _consumers_, _componentes de orden superior_, _render props_ y otras abstracciones. Aunque podemos [filtrarlos usando los DevTools](https://github.com/facebook/react-devtools/pull/503), esto apunta a un problema aún más profundo: React necesita una mejor primitiva para compartir lógica de estado.
```

File: content/docs/hooks-intro.md [hunk 9]
```diff
@@ -1,112 +1,112 @@
...
 
-With Hooks, you can extract stateful logic from a component so it can be tested independently and reused. **Hooks allow you to reuse stateful logic without changing your component hierarchy.** This makes it easy to share Hooks among many components or with the community.
+Con Hooks, puedes extraer lógica de estado de un componente y de tal forma este puede ser probado y re-usado independientemente. **Los Hooks te permiten reutilizar lógica de estado sin cambiar la jerarquía de tu componente.** Esto facilita el compartir Hooks entre muchos componentes o incluso con la comunidad.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
Con Hooks, puedes extraer lógica de estado de un componente de tal forma que este pueda ser probado independientemente y re-usado . **Los Hooks te permiten reutilizar lógica de estado sin cambiar la jerarquía de tu componente.** Esto facilita el compartir Hooks entre muchos componentes o incluso con la comunidad.
```

File: content/docs/hooks-intro.md [hunk 10]
```diff
@@ -1,112 +1,112 @@
...
 
-We'll discuss this more in [Building Your Own Hooks](/docs/hooks-custom.html).
+Discutiremos esto más a fondo en [Construyendo Tus Propios Hooks](/docs/hooks-custom.html).
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
Discutiremos esto más a fondo en [Construyendo tus propios Hooks](/docs/hooks-custom.html).
```

File: content/docs/hooks-intro.md [hunk 11]
```diff
@@ -1,112 +1,112 @@
...
+A menudo tenemos que mantener componentes que empiezan simples pero con el pasar del tiempo crecen y se convierten en un lío inmanejable de multiples lógicas de estado y efectos secundarios. Cada método del ciclo de vida a menudo contiene una mezcla de lógica no relacionada entre sí. Por ejemplo, los componentes pueden realizar alguna consulta de datos en el `componentDidMount` y `componentDidUpdate`. Sin embargo, el mismo método `componentDidMount` también puede contener lógica no relacionada que cree event listeners, y los limpie en el `componentWillUnmount`. El código relacionado entre sí que cambia a la vez está separado, pero el código que no tiene nada que ver termina combinado en un solo método. Esto hace que sea demasiado fácil introducir errores e inconsistencias.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
A menudo tenemos que mantener componentes que empiezan simples pero con el pasar del tiempo crecen y se convierten en un lío inmanejable de multiples lógicas de estado y efectos secundarios. Cada método del ciclo de vida a menudo contiene una mezcla de lógica no relacionada entre sí. Por ejemplo, los componentes pueden realizar alguna consulta de datos en el `componentDidMount` y `componentDidUpdate`. Sin embargo, el mismo método `componentDidMount` también puede contener lógica no relacionada que cree escuchadores de eventos, y los limpie en el `componentWillUnmount`. El código relacionado entre sí y que cambia a la vez es separado, pero el código que no tiene nada que ver termina combinado en un solo método. Esto hace que sea demasiado fácil introducir errores e inconsistencias.
```

File: content/docs/hooks-intro.md [hunk 12]
```diff
@@ -1,112 +1,112 @@
...
+Además de dificultar la reutilización y organización del código, hemos descubierto que las clases pueden ser una gran barrera para el aprendizaje de React. Tienes que entender cómo funciona `this` en JavaScript, que es muy diferente de cómo funciona en la mayoría de los lenguajes. Tienes que recordar agregar _bind_ a tus manejadores de eventos. Sin inestables [propuestas de sintaxis](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/), el código es muy verboso.  Las personas pueden entender _props_, el estado, y el flujo de datos de arriba hacia abajo perfectamente, pero todavía tiene dificultades con las clases. La distinción entre componentes de función y de clase en React y cuándo usar cada uno de ellos lleva a desacuerdos incluso entre los desarrolladores experimentados de React.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
Además de dificultar la reutilización y organización del código, hemos descubierto que las clases pueden ser una gran barrera para el aprendizaje de React. Tienes que entender cómo funciona `this` en JavaScript, que es muy diferente a cómo funciona en la mayoría de los lenguajes. Tienes que recordar agregar _bind_ a tus manejadores de eventos. Sin inestables [propuestas de sintaxis](https://babeljs.io/docs/en/babel-plugin-transform-class-properties/), el código es muy verboso.  Las personas pueden entender _props_, el estado, y el flujo de datos de arriba hacia abajo perfectamente, pero todavía tiene dificultades con las clases. La distinción entre componentes de función y de clase en React y cuándo usar cada uno de ellos lleva a desacuerdos incluso entre los desarrolladores experimentados de React.
```

File: content/docs/hooks-intro.md [hunk 13]
```diff
@@ -1,112 +1,112 @@
...
 
-To solve these problems, **Hooks let you use more of React's features without classes.** Conceptually, React components have always been closer to functions. Hooks embrace functions, but without sacrificing the practical spirit of React. Hooks provide access to imperative escape hatches and don't require you to learn complex functional or reactive programming techniques.
+Para resolver estos problemas, **Hooks te permiten usar más de las funciones de React sin clases.** Conceptualmente, los componentes de React siempre han estado más cerca de las funciones. Los Hooks abarcan funciones, pero sin sacrificar el espíritu práctico de React. Los Hooks proporcionan acceso a escotillas de escape imprescindibles y no requieren que tu aprendas técnicas complejas de programación funcional o reactiva.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
Para resolver estos problemas, **Hooks te permiten usar más de las funciones de React sin clases.** Conceptualmente, los componentes de React siempre han estado más cerca de las funciones. Los Hooks abarcan funciones, pero sin sacrificar el espíritu práctico de React. Los Hooks proporcionan acceso a vías de escape imprescindibles y no requieren que aprendas técnicas complejas de programación funcional o reactiva.
```

File: content/docs/hooks-intro.md [hunk 14]
```diff
@@ -1,112 +1,112 @@
...
 
->Examples
+>Ejemplos
 >
->[Hooks at a Glance](/docs/hooks-overview.html) is a good place to start learning Hooks.
+>[Hooks a Simple Vista](/docs/hooks-overview.html) es un buen lugar para comenzar a aprender Hooks.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
>[Hooks a simple vista](/docs/hooks-overview.html) es un buen lugar para comenzar a aprender Hooks.
```

File: content/docs/hooks-intro.md [hunk 15]
```diff
@@ -1,112 +1,112 @@
...
 
-## Gradual Adoption Strategy {#gradual-adoption-strategy}
+## Estrategia de adopción gradual {#gradual-adoption-strategy}
 
->**TLDR: There are no plans to remove classes from React.**
+>**TLDR: No hay planes para eliminar las clases de React.**
 
-We know that React developers are focused on shipping products and don't have time to look into every new API that's being released. Hooks are very new, and it might be better to wait for more examples and tutorials before considering learning or adopting them.
+Sabemos que los desarrolladores de React estan enfocados en la creación de productos de software y no tienen tiempo para analizar cada nueva API que se está lanzando. Los Hooks son muy nuevos, y tal vez sea mejor esperar más ejemplos y tutoriales antes de pensar en aprenderlos o adoptarlos.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
Sabemos que los desarrolladores de React están enfocados en la creación de productos de software y no tienen tiempo para analizar cada nueva API que se está lanzando. Los Hooks son muy nuevos, y tal vez sea mejor esperar más ejemplos y tutoriales antes de pensar en aprenderlos o adoptarlos.
```

File: content/docs/hooks-intro.md [hunk 16]
```diff
@@ -1,112 +1,112 @@
...
 
-We also understand that the bar for adding a new primitive to React is extremely high. For curious readers, we have prepared a [detailed RFC](https://github.com/reactjs/rfcs/pull/68) that dives into motivation with more details, and provides extra perspective on the specific design decisions and related prior art.
+También entendemos que la barra para añadir un nuevo primitivo a React es extremadamente alta. Para los lectores curiosos, hemos preparado un,[RFC detallado](https://github.com/reactjs/rfcs/pull/68) que se sumerge en la motivación con más detalles, y proporciona una perspectiva extra sobre las decisiones de diseño específicas y el estado de la técnica relacionado.
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
También entendemos que la barra para añadir una nueva primitiva a React es extremadamente alta. Para los lectores curiosos, hemos preparado un [RFC detallado](https://github.com/reactjs/rfcs/pull/68) que se sumerge en la motivación con más detalles y proporciona una perspectiva extra sobre las decisiones de diseño específicas y el estado de la técnica relacionado.
```

File: content/docs/hooks-intro.md (hunk 1 above)
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
**Es crucial, que los Hooks trabajen codo a codo con el código existente para que puedas adoptarlos gradualmente.** Estamos compartiendo esta API experimental para obtener retroalimentación temprana de aquellos en la comunidad que están interesados en dar forma al futuro de React - y vamos a iterar en Hooks de forma abierta.
```

File: content/docs/hooks-intro.md [hunk 17]
```diff
@@ -1,112 +1,112 @@
...
+## Preguntas frecuentes {#frequently-asked-questions}
 
-We've prepared a [Hooks FAQ page](/docs/hooks-faq.html) that answers the most common questions about Hooks.
+Hemos preparado una página de [Preguntas frecuentes acerca de Hooks](/docs/hooks-faq.html) que responde a las preguntas más frecuentes sobre Hooks.
 
-## Next Steps {#next-steps}
+## Próximos pasos {#next-steps}
 
-By the end of this page, you should have a rough idea of what problems Hooks are solving, but many details are probably unclear. Don't worry! **Let's now go to [the next page](/docs/hooks-overview.html) where we start learning about Hooks by example.**
+Al final de esta página, usted debe tener una idea aproximada de los problemas que los hooks están resolviendo, pero muchos detalles probablemente no están claros. ¡No te preocupes! **Ahora vayamos a la [página siguiente](/docs/hooks-overview.html) donde empezamos a aprender sobre los hooks por medio de un ejemplo.**
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
Al final de esta página, deberías tener una idea aproximada de los problemas que los Hooks están resolviendo, pero muchos detalles probablemente no están claros. ¡No te preocupes! **Ahora vayamos a la [siguiente página](/docs/hooks-overview.html) donde empezamos a aprender sobre los Hooks por medio de ejemplos.**
```

File: content/docs/hooks-intro.md [hunk 18]
```diff
@@ -1,112 +1,112 @@
...
+>Nota
 >
->React 16.8.0 is the first release to support Hooks. When upgrading, don't forget to update all packages, including React DOM. React Native will support Hooks in the next stable release.
+>React 16.8.0 es la primer versión que soporta Hooks. Al actualizar, no olvides actualizar todos los paquetes, incluyendo React DOM. React native soportará Hooks en la próxima versión estable.
 
-## Video Introduction {#video-introduction}
+## Video de introducción {#video-introduction}
 
-At React Conf 2018, Sophie Alpert and Dan Abramov introduced Hooks, followed by Ryan Florence demonstrating how to refactor an application to use them. Watch the video here:
+En el React Conf 2018, Sophie Alpert y Dan Abramov presentaron Hooks, Seguidos por Ryan Florence demostrando cómo refactorizar una aplicación usandolos. Mira el video aquí:
```
Comment (Darking360) [inline content/docs/hooks-intro.md]:
```suggestion
En el React Conf 2018, Sophie Alpert y Dan Abramov presentaron Hooks, seguidos por Ryan Florence demostrando cómo refactorizar una aplicación usándolos. Mira el video aquí:
```

**Comment by KevinGomezDev** [comment]
@Darking360 @icarlossz Fixes were added on commits d48b938 and a21f526, let me know if there's something else missing or if iit is ready to go.

**Comment by Darking360** [comment]
@KevinGomezDev Everything looks good for me. I just added a uppercase `n` for `React Native` and translate back to english the code example since we're doing those in english unless they are comments, I already pushed those so 🚀 Thanks again for the amazing work 🍺

**Pull request merged**